dotCMS/src/test/java/com/dotcms/util/pagination/SiteViewPaginatorUnitTest.java

```diff
@@ -0,0 +1,211 @@
+
+        //Then the rest.
+        final PaginatedArrayList<SiteView> itemsPage2 = paginator
+                .getItems(user, null, 100, limit + 1 , null, null, Collections.emptyMap());
+
+        final List<String> pageNamesPage2 = itemsPage2.stream().map(SiteView::getName).collect(Collectors.toList());
+        Assert.assertTrue(Ordering.<String> natural().isOrdered(pageNamesPage2));
+
+        //Test a page with both mixed configured and non-configured items are sorted.
+        limit = sitesWithIntegrations.size() + 2;
+        final PaginatedArrayList<SiteView> itemsPageMixed = paginator
+                .getItems(user, null, limit, 0, null, null, Collections.emptyMap());
+
+        final List<String> pageNamesConfiguredItemsPage = itemsPage2.stream().filter(SiteView::isConfigured).map(SiteView::getName).collect(Collectors.toList());
+        Assert.assertTrue(Ordering.<String> natural().isOrdered(pageNamesConfiguredItemsPage));
+
+        final List<String> pageNamesNonConfiguredItemsPage = itemsPage2.stream().filter(siteView -> !siteView.isConfigured()).map(SiteView::getName).collect(Collectors.toList());
+        Assert.assertTrue(Ordering.<String> natural().isOrdered(pageNamesNonConfiguredItemsPage));
+
+    }
+
+    private Host mockSite(final String identifier,final String name) {
+        final Host host = mock(Host.class);
+        when(host.getHostname()).thenReturn(name);
+        when(host.getName()).thenReturn(name);
+        when(host.getInode()).thenReturn(UUIDUtil.uuid());
+        when(host.getIdentifier()).thenReturn(identifier);
+        return host;
+    }
+
+    private User mockAdminUser() {
+        final User adminUser = mock(User.class);
+        when(adminUser.getUserId()).thenReturn("dotcms.org.1");
+        when(adminUser.getEmailAddress()).thenReturn("admin@dotcms.com");
+        when(adminUser.getFirstName()).thenReturn("Admin");
+        when(adminUser.getLastName()).thenReturn("User");
+        return adminUser;
+    }
+
+    private List<String> mockAllSitesIdentifiers(int allSitesNumber){
+        final List<String> allSites = new LinkedList<>();
+        allSites.add(0, Host.SYSTEM_HOST);
+        for(int i=0; i<= allSitesNumber; i++){
+            allSites.add(""+i);
+        }
+        return allSites;
+    }
+
+    private Set<String> mockSitesWithIntegrations(final List<String> allSites, final int bound){
+        if( bound > allSites.size()){
+           throw new IllegalArgumentException("bound must be less or equal to allSites.size ");
+        }
+        final Random random = new Random();
+        final List<String> sitesWithIntegrations = new LinkedList<>();
+        for (int i=0; i <= bound; i++ ){
+            sitesWithIntegrations.add(allSites.get(random.nextInt(bound)));
+        }
+        return new HashSet<>(sitesWithIntegrations);
+    }
+
+    private List<ContentletSearch> mockSearchResults(List<String> allSites){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'allSites' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4960082)